woocommerce/woocommerce-ios
# [Mobile Payments] Track `country` and `currency` properties in `payments_flow` collect and completed events

Closes: #11654 


## Why

From TTP analysis p1704815317432349-slack-C06C3CZFHGD, it'd be helpful if we have `country` and `currency` properties in `payments_flow_collect` and `payments_flow_completed` Tracks events. This PR ensures that these two properties are tracked in the two events. Other `payments_flow` events like `payments_flow_failed` and `payments_flow_canceled` will be added in a separate PR since they usage is more scattered.

## How

`payments_flow_completed` already tracks the currency, just the `country` prop was added. It uses `cardPresentPaymentsConfiguration.countryCode` for the country code, which is the store country from the WooCommerce settings.

## Testing instructions


- [x] @jaclync test the card reader payment flow
- [x] @jaclync tests a different currency/country
- [x] @jaclync adds the event properties to Tracks

Hand testing is optional, since it's pretty straightforward to verify but please feel free to test any edge cases as well.

Prerequisite: the store has at least one order that is eligible for payment

- Go to the orders tab
- Tap on the order in the prerequisite
- Tap `Collect Payment`
- Tap `Cash` as the easiest way to test --> `🔵 Tracked payments_flow_collect, properties: [AnyHashable("currency"): "USD", AnyHashable("country"): "US", ...]` should be shown in the console
- Tap `Mark as Paid` in the alert --> `🔵 Tracked payments_flow_completed, properties: [AnyHashable("currency"): "USD", AnyHashable("country"): "US", ...]` should be shown in the console


---
